lib/Doctrine/Migrations/Tools/Console/ConsoleInputMigratorConfigurationFactory.php

```diff
@@ -45,15 +45,14 @@ private function determineAllOrNothingValueFrom(InputInterface $input): bool|nul
                 <<<'DEPRECATION'
                     Context: Passing values to option `--all-or-nothing`
                     Problem: Passing values is deprecated
-                    Solution: If you need to disable the behavior, omit the option,
-                    otherwise, pass the option without a value
+                    Solution: how we should deal with this option being `true` in config but user want to disable it in command line for 4.x?
```

Comment:
Added intentionally this line, since I'm wondering how to handle this.

Also as an extra thought, since we've added more coverage to how the option is determined, I wonder if it still makes sense to deprecate this ability, or if we still want to go ahead with disallowing values to be passed for this option.

Reply:
What about a new `--no-all-or-nothing` option?

Reply:
> Also as an extra thought, since we've added more coverage to how the option is determined, I wonder if it still makes sense to deprecate this ability, or if we still want to go ahead with disallowing values to be passed for this option.

I don't think there's much value in having several ways of doing the same thing. It just adds complexity.

> What about a new --no-all-or-nothing option?

Might be good, and is easy to implement with https://symfony.com/blog/new-in-symfony-5-3-negatable-command-options

Reply:
@greg0ire that's a good idea, but it seems for our particular use case it won't work. looking at the docs:

> Negatable options are only available for options that don't allow passing any value to them (their previous type should be InputOption::VALUE_NONE).

For the remainder of the 3.x lifetime, it is better the users continue passing the optional value as it's currently allowed, and we introduce the negatable option in 4.x.

Reply:
Maybe we can use a negatable option in 4.x, and a full-fledged option in 3.x, literally named `no-all-or-nothing` to help with the transition?